always define PATCH. without it, puppetd hangs when upgrading INTERACTIVE port.
http://reductivelabs.com/trac/puppet/ticket/624
--- lib/puppet/provider/package/ports.rb.orig	Tue May  8 14:55:34 2007
+++ lib/puppet/provider/package/ports.rb	Tue May  8 14:55:57 2007
@@ -19,7 +19,7 @@
         # -p: create a package
         # -N: install if the package is missing, otherwise upgrade
         # -P: prefer binary packages
-        cmd = %w{-p -N -P} << @model[:name]
+        cmd = %w{-p -N -P -M BATCH=yes} << @model[:name]
 
         output = portupgrade(*cmd)
         if output =~ /\*\* No such /
